Fix - Wrong value shown in the filter modal

kamilmielnik opened the pull request:
Fixes #48319

### Description

The issue prevailed for [all types of filters except date](https://github.com/metabase/metabase/blob/10bf40c323d921eba3663f9eb0234acdbf44b67f/frontend/src/metabase/querying/filters/components/FilterModal/ColumnFilterSection/ColumnFilterSection.tsx#L40-L58) (i.e. booleans, coordinates, "default", number, string, time).
It was not only about the filter values, but also the filter operator.

### How to verify

Follow repro steps for all types of filters except date: booleans, coordinates, "default", number, string, time.

### Demo

Before:

https://github.com/user-attachments/assets/00185a5f-9a9c-4105-aa4a-0313f9bdef53

After:

https://github.com/user-attachments/assets/31cd5de0-88d0-470e-b509-b7dbe2caf33e

kamilmielnik committed:
Sync state in useStringFilter

kamilmielnik committed:
Sync state in useBooleanOperatorFilter

kamilmielnik committed:
Sync state in useBooleanOptionFilter

kamilmielnik committed:
Add FilterModalResult type

kamilmielnik committed:
Fix setTab type

kamilmielnik committed:
Make comments work as JSDocs

kamilmielnik committed:
Move FilterModalResult

kamilmielnik committed:
Add filter modal context

kamilmielnik committed:
Drop FilterModalHeader and use FilterSearchInput directly

kamilmielnik committed:
Start to use context in BooleanFilterEditor

kamilmielnik committed:
Start to use context in TimeFilterEditor

kamilmielnik committed:
Start to use context in DateFilterEditor

kamilmielnik committed:
Start to use context in CoordinateFilterEditor

kamilmielnik committed:
Start to use context in NumberFilterEditor

kamilmielnik committed:
Start to use context in StringFilterEditor

kamilmielnik committed:
Start to use context in DefaultFilterEditor

kamilmielnik committed:
Stop drilling isSearching

kamilmielnik committed:
Fix tests

kamilmielnik committed:
Stop drilling version prop

kamilmielnik committed:
Remove FilterModalBody and FilterModalFooter exports

kamilmielnik committed:
Make naming consistent

kamilmielnik committed:
Remove props drilling

kamilmielnik committed:
Remove props drilling

kamilmielnik committed:
Remove props drilling

kamilmielnik committed:
Inline FilterTabContent into FilterModalBody

kamilmielnik committed:
Revert "Inline FilterTabContent into FilterModalBody"

This reverts commit cfd5738e52d5e45f9025d538589363e0cf1ed78a.

kamilmielnik committed:
Remove props drilling

kamilmielnik committed:
Sort attributes

kamilmielnik committed:
Consolidate imports

kamilmielnik committed:
Fix a mistake

kamilmielnik committed:
Improve JSDoc

kamilmielnik committed:
Merge branch 'master' into 48319-refactor-filter-modal

kamilmielnik committed:
Revert "Sync state in useBooleanOptionFilter"

This reverts commit 33a979dfdf92070e356d85acb84f860b8c7cc002.

kamilmielnik committed:
Merge branch '48319-refactor-filter-modal' into 48319-filter-modal

kamilmielnik committed:
Update approach after merging refactoring branch

kamilmielnik committed:
Add a repro for string filters

kamilmielnik committed:
Go back and forth on search input

kamilmielnik committed:
Assert operator

kamilmielnik committed:
Move effect to concrete hook

kamilmielnik committed:
Add test for boolean filters

kamilmielnik committed:
Fix time filters

kamilmielnik committed:
Revert "Move effect to concrete hook"

This reverts commit 82b94f4345204a1201cdaf6642782c2cbd8abd94.

kamilmielnik committed:
Move effect outside use-time-filter

kamilmielnik committed:
Improve naming

kamilmielnik committed:
Remove TODO

kamilmielnik committed:
Extract setup48319

kamilmielnik committed:
Rename setup48319 to setupCustomColumns

kamilmielnik committed:
Rename handle* to on*

kamilmielnik committed:
Merge branch 'master' into 48319-refactor-filter-modal

kamilmielnik committed:
Merge branch '48319-refactor-filter-modal' into 48319-filter-modal

kamilmielnik committed:
Fix coordinate filters

kamilmielnik committed:
Merge branch 'master' into 48319-refactor-filter-modal

kamilmielnik committed:
Merge branch '48319-refactor-filter-modal' into 48319-filter-modal

kamilmielnik committed:
Fix default filters

kamilmielnik committed:
Fix number filters

ranquild reviewed (commented):
@kamilmielnik wdyt about using `searchText` for `key`?

kamilmielnik commented:
> @kamilmielnik wdyt about using `searchText` for `key`?

I personally consider this a hack and try to avoid it.
It would make this PR a ton shorter though.

@ranquild WDYT about it? :)

ranquild commented:
@kamilmielnik I think if we could use it somewhere higher in the tree it would be okay. With a basic test that it works

kamilmielnik commented:
> @kamilmielnik I think if we could use it somewhere higher in the tree it would be okay. With a basic test that it works

I'll look into it

kamilmielnik committed:
Revert 1st approach

kamilmielnik committed:
Merge branch 'master' into 48319-filter-modal

kamilmielnik committed:
Use key prop

kamilmielnik committed:
Remove detailed tests and leave only one

ranquild commented:
@kamilmielnik actually take a look at this https://github.com/metabase/metabase/blob/a7b8f79b031eefeff5b3b0c8f8f40bcb7fd2e005/frontend/src/metabase/querying/filters/hooks/use-filter-modal/use-filter-modal.ts#L50

kamilmielnik committed:
Merge branch 'master' into 48319-filter-modal

kamilmielnik committed:
Use current value

kamilmielnik committed:
Add a test for "version" hack

kamilmielnik committed:
Use only version for the key

kamilmielnik committed:
Rename version to remountKey and add explanatory comments

kamilmielnik committed:
Rename forceDescendantsRemount to forceRemountDescendants

kamilmielnik commented:
> @kamilmielnik actually take a look at this
> 
> https://github.com/metabase/metabase/blob/a7b8f79b031eefeff5b3b0c8f8f40bcb7fd2e005/frontend/src/metabase/querying/filters/hooks/use-filter-modal/use-filter-modal.ts#L50
> 
> We should probably increment the version key when search changes

Ah, great catch!

Given how [_astonished_](https://en.wikipedia.org/wiki/Principle_of_least_astonishment) I am by this, I have:
- renamed `version` to `remountKey` to make the purpose more apparent
- sprinkled some comments on top
- added a test for the existing `key` hack which kicked in when clearing all filters

kamilmielnik committed:
Rename
